go.mod

```diff
@@ -4,39 +4,43 @@ go 1.23.2
 
 require (
 	github.com/go-openapi/strfmt v0.23.0
-	github.com/go-openapi/swag v0.22.3
+	github.com/go-openapi/swag v0.23.0
 	github.com/go-test/deep v1.1.1
 	github.com/hashicorp/go-cty v1.4.1-0.20200414143053-d3edf31b6320
 	github.com/hashicorp/terraform-plugin-sdk/v2 v2.16.0
 	github.com/pkg/errors v0.9.1
 	github.com/stretchr/testify v1.10.0
 	go.pinniped.dev v0.23.0
-	golang.org/x/exp v0.0.0-20221126150942-6ab00d035af9
-	k8s.io/apiextensions-apiserver v0.28.2
-	k8s.io/apimachinery v0.28.2
-	k8s.io/client-go v0.28.2
-	sigs.k8s.io/controller-runtime v0.6.0
+	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56
+	k8s.io/apiextensions-apiserver v0.32.0
+	k8s.io/apimachinery v0.32.0
+	k8s.io/client-go v0.32.0
+	sigs.k8s.io/controller-runtime v0.20.0
```

Comment:
I am a bit hesitant on this without running some tests. Controller runtime has been known to create issues sometimes between versions and this is a long jump